aya/src/bpf.rs

```diff
@@ -155,6 +156,36 @@ impl<'a> BpfLoader<'a> {
         self
     }
 
+    /// Sets the value of a global variable
+    ///
+    /// From Rust eBPF, a global variable would be constructed as follows:
+    /// ```no run
+    /// #[no_mangle]
+    /// const VERSION = 0;
+    /// ```
+    /// If using a struct, ensure that it is `#[repr(C)]` to ensure the size will
+    /// match that of the corresponding ELF symbol.
+    ///
+    /// From C eBPF, you would annotate a variable as `volatile const`
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use aya::BpfLoader;
+    ///
+    /// let bpf = BpfLoader::new()
+    ///     .set_global("VERSION", &2)
+    ///     .load_file("file.o")?;
+    /// # Ok::<(), aya::BpfError>(())
+    /// ```
+    ///
+    pub fn set_global<V: Pod>(&mut self, name: &'a str, value: &'a V) -> &mut BpfLoader<'a> {
```

Comment:
I don't love that we need to pass V by reference here. One way to avoid it would be by adding a `as_bytes()` default trait method to the `Pod` trait, which is something that I've been thinking about for a while. Then we could store V in the hash map, and call `v.as_bytes()` when needed.

Anyway I can do this in a followup PR.

Reply:
I did look at that, but it seemed like a breaking change. Anyone that's implemented the `Pod` trait for a Map K/V will have to implement `as_bytes` also. It may make sense to make the `unsafe_impl_pod` macro public to cover the default case, or perhaps use a proc_macro so we can use can `#[derive(Pod)]` instead.

Even better if bindgen could also add that derive to C structs :tada:

Reply:
Traits can have default method implementations, and implementors don't have to implement those. So we could implement it on trait Pod and that's it

Reply:
>  Then we could store V in the hash map, and call v.as_bytes() when needed.

The problem is that `Pod` is not object-safe since it's `Copy + `static` :cry: